### ISSUE TITLE
ASYNC102 await-in-finally-or-cancelled false alarms on nurseries

### ISSUE THREAD

**Issue by jakkdl** [python-trio/flake8-async#283]
https://github.com/python-trio/flake8-async/blob/225f15a98db4bb8ce4600127fd0fe90bc5c9c04b/tests/eval_files/async102.py#L117-L125

TODO:
* [ ] Special-case `async with open_nursery(...)`, and `async with anyio.create_task_group(...)` as not being checkpoints. (also affects other rules)
* [ ] handle `[nursery_name].cancel_scope.[deadline/shield]`

**Comment by jakkdl** [comment]
The other rules that could be relevant are:
* ASYNC100 cancel-scope-no-checkpoint
  * A cancel scope with only a nursery can never be cancelled, so this should ignore `open_nursery`/`create_task_group`
* ASYNC120 await-in-except
  * Same as 102, and also already fixed by #289 
* ASYNC91x
  * nurseries can checkpoint, so these should not ignore nurseries, and do not need to be changed.

```python
async def foo(): # should not give ASYNC910
  async with trio.open_nursery() as nursery():
    nursery.start_soon(bar)
```

tl;dr, ASYNC100 also needs changing

**Comment by jakkdl** [comment]
Hrm, neither of these statements are cancellation sources
```python
my_nursery = await trio.open_nursery()
async with my_nursery:
    ...
```
So it's not just `async with trio.open_nursery()` that needs to be checked against.
Both of those cases are definitely less common, but far from unheard of. The former is an easy fix, the latter is a bit more tricky but we're already type-tracking nurseries for other reasons so shouldn't be too bad.
